### ISSUE TITLE
(all) content items should have ownership property (multi-site editor) 

### ISSUE THREAD

**Issue by MyPyDavid** [rdmorganiser/rdmo#490]
### Rationale / Begründung

An editor of a tenant instance as part of a multi-tenant rdmo instance wants to make sure that the content items, which were created by the tenant, can only be edited by themselves and colleagues.

All content items (attributes, optionsets, questions etc.) are global, i.e. every editor of an instance is able to change any content item. 
In multi-site mode this means that any editor of a tenant can change the content of another tenant.
To prevent this, an additional property for every content item "owner" should be added. Owner of an item are sites (not users), that means editors of one site can change all items that they created. Only owners can edit and update the content items. 
Additionally, there might be multi-site editors and administrators who can edit all items. Non-owners can view items, use them and copy them, but cannot change them. 
There may be items that are not owned by any site, but globally used for every site. These items can only be changed by multi-site editors and administrators.

### Affected 

Managers, Admins

#### Model field and permissions
* a Field ~~"owner"~~ is added to every content item Model (e.g. ManyToManyField to Sites)
  * `editor` , as in "content item editorial site", or "site ownership"?
* the ownership can be declared as "global", which means no site editors are allowed to edit the item (except super editor and administrators)
  * special permission when the field `editor` is empty
* multi-site editors, administrators can edit the ownership of an item
  *  need to check the permissions
 
#### CRUD Behaviour
* when a user creates a new content item, the owner is set to the site of which the user is a member of
  * can added via the `current_site()` of the User
* the property is not part of the export of content items (catalog etc.)
  * only for internal CMS purposes
* the owner of an item is mentioned in the update content item details form
  * another field in the `modal form` 
* if a non-owner tries to change an owned item, a error message is shown stating the owner of the item
  * similar to the `is_locked` functionality

**Comment by jochenklar** [comment]
`owner` would be confusing with the `owner` of projects, maybe `editors` would be better (being a m2m field to site).

**Comment by MyPyDavid** [comment]
For adding the DRF [Object level permissions](https://www.django-rest-framework.org/tutorial/4-authentication-and-permissions/#object-level-permissions) to the `Catalog`. 
I could add this `IsEditorOrReadOnly` class to the `permission_classes` of the `CatalogViewSet`.  Then I don't need to use the `rules` and the `HasObjectPermission`  classes.


```py
class IsEditorOrReadOnly(BasePermission):
    """
    Object-level permission to only allow owners of an object to edit it.
    Assumes the model instance has an `editors` attribute.

    """
    
    def has_object_permission(self, request, view, obj):
        
        # Read permissions are allowed to any request,
        # so we'll always allow GET, HEAD or OPTIONS requests.
        if request.method in SAFE_METHODS:
            return True
        # Instance must have an attribute named `editors`.        
        return request.user.role.editor.filter(id__in=obj.editors.all()).exists()
```
and give this to the `CatalogViewSet`
```py
class CatalogViewSet(CopyModelMixin, ModelViewSet):
    permission_classes = (HasModelPermission & IsEditorOrReadOnly, )
    ....
```

Would this way be sufficient or does it need to be implemented via `rules` and `HasObjectPermission` ?

**Comment by MyPyDavid** [comment]
can be closed by [2.0.0](https://github.com/rdmorganiser/rdmo/releases/tag/2.0.0)

after the upgrade, we assigned sites to the element's editors by a mapping based on the uri-prefixes.
 It was executed by a costum django management command in the rdmo-app
